flashinfer/utils.py

```
@@ -121,6 +121,18 @@ def next_positive_power_of_2(x: int) -> int:
     return n + 1
 
 
+def last_positive_power_of_2(x: int) -> int:
+    """Return the largest power of 2 that is ``<= x``.
+
+    If *x* is itself a power of 2, returns *x*.
+    """
+    next = next_positive_power_of_2(x)
```

Ruff (0.15.17)

[error] Variable `next` is shadowing a Python builtin

(A001)